beginner_source/Pretraining_Vgg_from_scratch.py

```diff
@@ -0,0 +1,665 @@
+"""
+``Pretraining`` VGG from scratch 
+============================
+
+
+**Author:** `WoongJoon Choi <https://github.com/woongjoonchoi>`_
+
+In this tutorial, we will embark on an exciting journey to build and
+train a VGG network from scratch using Python and popular deep learning
+libraries such as PyTorch. We will dive into the details of the VGG
+architecture, understanding its components and the rationale behind its
+design.
+
+Our tutorial is designed for both beginners who are new to deep learning
+and seasoned practitioners looking to deepen their understanding of CNN
+architectures.
+
+Before you start 
+
+.. code-block:: sh
+
+    pip install albumentations
+"""
+import subprocess
+import sys
+
+try:
+    import albumentations
+    print("albumentations are already installed")
+except ImportError:
+    print("albumentations module not found. Installing...")
+    subprocess.check_call([sys.executable, "-m", "pip", "install", "albumentations"])
+    print("albumentations module installed successfully.")
+
+
+
+import torch.optim as optim
+import albumentations as A
+import numpy as np
+import torch
+
+from torchvision.datasets import CIFAR100,CIFAR10,MNIST,ImageNet
+import os
+from PIL import Image
+
+
+
+
+
+######################################################################
+# We recommend using GPU for this tutorial.
+# 
+
+device = 'cuda' if torch.cuda.is_available() else 'cpu'
+
+
+######################################################################
+# Worth point of this tutorial
+# ----------------------------
+# 
+
+
+######################################################################
+# -  We train the model from scratch using only the configuration
+#    presented in the paper.
+# 
+#    -  we do not use future method, like ``Batch normalization``,Adam , He
+#       initialization.
+# 
+# -  You can apply to ImageNet Data.
+# 
+#    -  If you can download the ImageNet Data(140GB), you can apply this
+#       tutorial to reproduce Original VGG.
+# 
+# -  You can learn VGG within the training time suggested in the paper.
+# 
+
+
+######################################################################
+# Why VGG is so popular ?
```

Comment:
I suggest renaming this section to `Background` and possibly beefing up the history behind VGG a tad bit more. Specifically, what was the core innovation that it introduced to become the new SotA? (e.g. deep layers of smaller, consistent 3x3 kernel convolutions replacing the variable-sized filters from AlexNet).